Source/BestProjectEver/Public/HUD/Widgets/BPE_CharacterOverlay.h

```diff
@@ -11,6 +11,33 @@ class UImage;
 class UProgressBar;
 class UTextBlock;
 class UBPE_TimerWidget;
+class UBPE_CrosshairWidget;
+
+
+USTRUCT()
+struct FWeaponIcon
+{
+	GENERATED_BODY()
+
+	UPROPERTY(Transient)
+	TObjectPtr<UImage> Icon;
+	
+	FLinearColor IconColor;
+
+	/** render scale when a weapon icon is selected */
+	UPROPERTY(EditDefaultsOnly, Category="Weapon")
+	FVector2D SelectedScale;
+
+	/** render scale when a weapon icon is not selected, by default in the widget blueprint the icons are set to this scale */
+	UPROPERTY(EditDefaultsOnly, Category="Weapon")
+	FVector2D NotSelectedScale;
+	
+	FWeaponIcon() : Icon(nullptr), IconColor(FLinearColor::Transparent), SelectedScale(FVector2D(2.0f, 2.0f)),
+		NotSelectedScale(FVector2D(0.8f, 0.8f)) {};
```

Comment:
Perfectly fine as it is, but just so you know, Vectors have a constructor where if you just set one float, it will assign that float to all of its components, so `FVector2D(0.8f)` is the same as `FVector2D(0.8f, 0.8f)` but more concise :)